tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -1,610 +1,527 @@
 import * as React from 'react';
-import { shallow } from 'enzyme';
+// import { within } from '@testing-library/react';
 
-import {
-  createInternalReview,
-  fetchUserReviews,
-  setUserReviews,
-} from 'amo/actions/reviews';
-import AddonsByAuthorsCard from 'amo/components/AddonsByAuthorsCard';
-import AddonReviewCard from 'amo/components/AddonReviewCard';
-import UserProfile, { extractId, UserProfileBase } from 'amo/pages/UserProfile';
-import NotFoundPage from 'amo/pages/ErrorPages/NotFoundPage';
-import UserProfileHead from 'amo/components/UserProfileHead';
-import ReportUserAbuse from 'amo/components/ReportUserAbuse';
+// import { setUserReviews } from 'amo/actions/reviews';
+// import { fetchUserReviews, setUserReviews } from 'amo/actions/reviews';
+import UserProfile, { extractId } from 'amo/pages/UserProfile';
 import {
   fetchUserAccount,
-  getCurrentUser,
+  // getCurrentUser,
   loadUserAccount,
 } from 'amo/reducers/users';
-import { DEFAULT_API_PAGE_SIZE, createApiError } from 'amo/api';
-import Paginate from 'amo/components/Paginate';
-import {
-  ADDON_TYPE_EXTENSION,
-  ADDON_TYPE_STATIC_THEME,
-  CLIENT_APP_FIREFOX,
-  USERS_EDIT,
-} from 'amo/constants';
-import { ErrorHandler } from 'amo/errorHandler';
-import { sendServerRedirect } from 'amo/reducers/redirectTo';
-import ErrorList from 'amo/components/ErrorList';
-import Icon from 'amo/components/Icon';
-import LoadingText from 'amo/components/LoadingText';
-import Rating from 'amo/components/Rating';
-import Page from 'amo/components/Page';
-import UserAvatar from 'amo/components/UserAvatar';
+// import { DEFAULT_API_PAGE_SIZE } from 'amo/api';
+// import { DEFAULT_API_PAGE_SIZE, createApiError } from 'amo/api';
+import { CLIENT_APP_FIREFOX, USERS_EDIT } from 'amo/constants';
+// import { ErrorHandler } from 'amo/errorHandler';
+// import { sendServerRedirect } from 'amo/reducers/redirectTo';
 import {
-  createStubErrorHandler,
+  // createFakeLocation,
+  createHistory,
+  // createStubErrorHandler,
   createUserAccountResponse,
   dispatchClientMetadata,
   dispatchSignInActions,
-  fakeI18n,
-  fakeReview,
-  createFakeLocation,
-  shallowUntilTarget,
+  // fakeReview,
+  getByClassName,
+  getByTagName,
+  // getElement,
+  renderPage as defaultRender,
+  screen,
 } from 'tests/unit/helpers';
 
 describe(__filename, () => {
+  const lang = 'fr';
+  const clientApp = CLIENT_APP_FIREFOX;
+  let store;
+
+  beforeEach(() => {
+    store = dispatchClientMetadata({ lang }).store;
+  });
+
   function defaultUserProps(props = {}) {
     return {
-      display_name: 'Matt MacTofu',
+      display_name: 'Display McDisplayNamey',
       userId: 500,
       ...props,
     };
   }
 
-  function signInUserWithProps({ userId = 123, ...props }) {
+  function signInUserWithProps({ userId = 123, ...props } = {}) {
+    store = dispatchSignInActions({
```

Comment:
The problem with the `lang` is multi-layered. 

Firstly, `dispatchSignInActions` creates a new `store`, so it loses everything from your existing `store`, which includes the `lang`.  So even though you are passing `store` into `dispatchSignInActions`, it doesn't actually do anything with it.

Secondly, `dispatchSignInActions` calls `dispatchClientMetadata`, passing in the args it receives when it is called, and `dispatchClientMetadata` has default values for a number of arguments, including `lang`, so, because you do not explicitly pass `lang` into *this* call to `dispatchSignInActions`, it uses the default `lang`, which is not the same one you use in your test.

A simple solution to this would be to pass `lang` into `dispatchSignInActions` where you call it from within `signInUserWithProps`, but I think it would be better if we could simply pass a `store` into `dispatchSignInActions`, and if we do that then it uses the passed-in `store` and doesn't overwrite any of the values. We should change `dispatchSignInActions` to allow for this behaviour, as it's consistent with us having a global `store` in our test suite.